### ISSUE TITLE
mesa build error in CentOS6: dependency and undefined "setUnsafeAlgebra()"

### ISSUE THREAD

**Issue by Gwang-Jin** [maxim-belkin/homebrew-xorg#374]
Hello.

Although `glproto` is installed in my system, when building `mesa`, I met the error messages
````
configure: error: Package requirements (glproto >= 1.4.14) were not met:

No package 'glproto' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables GLPROTO_CFLAGS
and GLPROTO_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
````

And 
````
gallivm/lp_bld_misc.cpp:833:13: error: 'class llvm::FastMathFlags' has no member named 'setUnsafeAlgebra'
       flags.setUnsafeAlgebra();
             ^
````

Thank you.

[https://gist.github.com/091234dc707b47899d6fda3f19b8f2bf](https://gist.github.com/091234dc707b47899d6fda3f19b8f2bf)
- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

To help us debug your issue please explain:
- What you were trying to do (and why)
- What happened (include command output)
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

**Comment by dpo** [comment]
`mesa` doesn't build at all. It's missing several dependencies. Currently, I've added
```diff
diff --git a/Formula/mesa.rb b/Formula/mesa.rb
index c0e705b..62543f6 100644
--- a/Formula/mesa.rb
+++ b/Formula/mesa.rb
@@ -18,6 +18,9 @@ class Mesa < Formula
   depends_on "bison" => :build
   depends_on "libtool" => :build

+  depends_on "linuxbrew/xorg/damageproto"
+  depends_on "linuxbrew/xorg/dri2proto"
+  depends_on "linuxbrew/xorg/glproto"
   depends_on "linuxbrew/xorg/libdrm"
   depends_on "systemd" # provides libudev <= needed by "gbm"
   depends_on "linuxbrew/xorg/libsha1"
``` 
but the build still fails with
```
Last 15 lines from /home/mgi.polymtl.ca/doorba/.cache/Homebrew/Logs/mesa/03.make:
libtool: link: ranlib .libs/libgalliumvl.a
libtool: link: ( cd ".libs" && rm -f "libgalliumvl.la" && ln -s "../libgalliumvl.la" "libgalliumvl.la" )
gallivm/lp_bld_misc.cpp: In function 'LLVMOpaqueBuilder* lp_create_builder(LLVMContextRef, lp_float_mode)':
gallivm/lp_bld_misc.cpp:833:13: error: 'class llvm::FastMathFlags' has no member named 'setUnsafeAlgebra'
       flags.setUnsafeAlgebra();
             ^
make[4]: *** [gallivm/lp_bld_misc.lo] Error 1
make[4]: Leaving directory `/tmp/mesa-20180605-9622-1m28ai0/mesa-17.2.3/src/gallium/auxiliary'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/tmp/mesa-20180605-9622-1m28ai0/mesa-17.2.3/src/gallium'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/tmp/mesa-20180605-9622-1m28ai0/mesa-17.2.3/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/tmp/mesa-20180605-9622-1m28ai0/mesa-17.2.3/src'
make: *** [all-recursive] Error 1
```
#351 makes some updates but it seems to have been abandoned.

**Comment by Gwang-Jin** [comment]
Yes. I met the same error message regarding `setUnsafeAlgebra()`.
But when I tried to find the definition of `setUnsafeAlgebra()`, there was no definition of `setUnsafeAlgebra()` anywhere. It seemed to be an error of mesa.

**Comment by nir-krakauer** [comment]
My build of `mesa` in Ubuntu fails with a different message:
`ImportError: No module named setuptools`
https://gist.github.com/1a74473593d7c0bd6d675c43f0cd9822

**Comment by Gwang-Jin** [comment]
@nir-krakauer Did you try to install `setuptools` using brewed `pip`?

`pip install setuptools`

**Comment by nir-krakauer** [comment]
`pip3 install setuptools` returns:

`Requirement already satisfied: setuptools in ./.linuxbrew/lib/python3.6/site-packages (39.2.0)`

**Comment by Gwang-Jin** [comment]
@nir-krakauer Could you please run not `pip3` but `pip2`?

**Comment by nir-krakauer** [comment]
Thanks, `brew reinstall python@2` did fix the problem.

**Comment by nir-krakauer** [comment]
Now I also get 
`configure: error: Package requirements (glproto >= 1.4.14) were not met:
No package 'glproto' found`
https://gist.github.com/d9620a94307e088d8a93642ba2b4b573

**Comment by Gwang-Jin** [comment]
@nir-krakauer Yes. As a recommendation from @dpo,

`brew edit mesa`

adding to mesa.rb
````
  depends_on "linuxbrew/xorg/damageproto"
  depends_on "linuxbrew/xorg/dri2proto"
  depends_on "linuxbrew/xorg/glproto"
````
. Then you would meet the error message we met.

**Comment by Gwang-Jin** [comment]
I found `setUnsafeAlgebra()` as member function of `class llvm::FastMathFlags` in `llvm-3.9`. And this is not member function in `llvm-6.0`. So this seems to be a problem of `llvm` version.

**Comment by dpo** [comment]
If that's the case, you should be able to declare something like
```
depends_on "llvm@3.9"
```

**Comment by maxim-belkin** [comment]
good catch @Gwang-Jin!

**Comment by Gwang-Jin** [comment]
In `llvm@3.9`, `llvm@4`, and `llvm@5`, `setUnsafeAlgebra()` is defined as a member function of `class llvm::FastMathFlags`. Only in `llvm@6` it is deleted.
Now `brew install llvm@3.9` seems to have an issue:
````
Undefined reference to __cxa_guard_acquire and __cxa_guard_release
````

**Comment by Gwang-Jin** [comment]
I succeeded with `llvm@4` at installing `mesa`.

**Comment by dpo** [comment]
Could you please post your diff (or submit a PR)?

**Comment by Gwang-Jin** [comment]
Ok. I submitted [#375](https://github.com/Linuxbrew/homebrew-xorg/pull/375/commits/f140b2e890cc5ebc72257583f3e495444e81fe2c).

**Comment by maxim-belkin** [comment]
This issue has been resolved in Linuxbrew/homebrew-xorg#375
